src/main/java/com/duckblade/osrs/dpscalc/calc/gearbonus/VampyreGearBonus.java

```diff
@@ -1,33 +1,60 @@
 package com.duckblade.osrs.dpscalc.calc.gearbonus;
 
+import com.duckblade.osrs.dpscalc.calc.WeaponComputable;
 import com.duckblade.osrs.dpscalc.calc.compute.ComputeContext;
 import com.duckblade.osrs.dpscalc.calc.compute.ComputeInputs;
+import com.duckblade.osrs.dpscalc.calc.model.DefenderAttributes;
 import com.duckblade.osrs.dpscalc.calc.model.GearBonuses;
+import com.google.common.collect.ImmutableMap;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.ItemID;
+
 import javax.inject.Inject;
 import javax.inject.Singleton;
-import lombok.RequiredArgsConstructor;
+import java.util.Map;
+
+import static com.duckblade.osrs.dpscalc.calc.compute.ComputeInputs.ATTACK_STYLE;
 
-// TODO
 @Singleton
 @RequiredArgsConstructor(onConstructor = @__(@Inject))
 public class VampyreGearBonus implements GearBonusComputable
 {
 
+	private static final Map<Integer, GearBonuses> weaponToBonus = ImmutableMap.of(
+			ItemID.IVANDIS_FLAIL, GearBonuses.of(1, 1.20),
+			ItemID.BLISTERWOOD_SICKLE, GearBonuses.of(1.05, 1.15),
+			ItemID.BLISTERWOOD_FLAIL, GearBonuses.of(1.05, 1.25)
+	);
+
+	private final WeaponComputable weaponComputable;
+
 	@Override
 	public boolean isApplicable(ComputeContext context)
 	{
-		if (context.get(ComputeInputs.DEFENDER_ATTRIBUTES).isVampyre())
+		if (!context.get(ComputeInputs.DEFENDER_ATTRIBUTES).isVampyre())
 		{
-			context.warn("Bonuses for vampyre immunities / vampyrebane is not yet implemented.");
-			return true;
+			return false;
 		}
 
-		return false;
+		if (!context.get(ATTACK_STYLE).getAttackType().isMelee())
+		{
+			return false;
+		}
+
+		int weapon = context.get(weaponComputable).getItemId();
+		return weaponToBonus.containsKey(weapon);
 	}
 
 	@Override
 	public GearBonuses compute(ComputeContext context)
 	{
-		return GearBonuses.EMPTY;
+		DefenderAttributes attributes = context.get(ComputeInputs.DEFENDER_ATTRIBUTES);
+		if (!attributes.isVampyre())
```

Comment:
This will never be hit since the `isApplicable` would guard against this. You can include the warning in the `isApplicable` by checking for weapon first, then vampyre.